event_model/__init__.py

```diff
@@ -1045,6 +1045,73 @@ def unpack_datum_page(datum_page):
         yield datum
 
 
+def rechunk_event_pages(event_pages, chunk_size):
+    remainder = chunk_size
+    chunk_list = []
+
+    for page in event_pages:
+        new_chunks = page_chunks(page, chunk_size, remainder)
+        for chunk in new_chunks:
+            remainder -= len(chunk['uid'])
+            chunk_list.append(chunk)
+            if remainder == 0:
+                yield merge_event_pages(chunk_list)
+                remainder = chunk_size
+    if chunk_list:
+        yield merge_event_pages(chunk_list)
+
+    def page_chunks(page, chunk_size, remainder):
+        array_keys = ['seq_num', 'time', 'uid']
+        page_size = len(page['uid'])
+        chunks = [(0,remainder)]
+        chunks.extend([(i - chunk_size, i) for i
+                       in range(remainder + chunk_size, page_size, chunk_size)]
+
+        for start, stop in chunks:
+            yield {**{'descriptor': page['descriptor']},
+                **{key: page[key][start:stop] for key in array_keys},
+                **{'data': page['data'][key][start:stop]
+                   for key in page['data'].keys()}.values())},
+                **{'timestamps': page['timestamps'][key][start: stop]
+                   for key in page['data'].keys()}.values())},
+                **{'filled': page['filled'][key][start:stop]
+                   for key in page['data'].keys()}.values())}}
+
+
+def merge_event_pages(pages):
+    if len(pages) == 1:
+        return pages[0]
+
+    array_keys = ['seq_num', 'time', 'uid']
+
+    return {**{'descriptor': pages[0]['descriptor']},
+            **{key: list(itertools.chain.from_iterable(
+                    [page[key] for page in pages]) for key in array_keys])},
+            **{'data': {key: list(itertools.chain.from_iterable(
```

Comment:
...But here and below, I don't think `**` is buying you anything. `{..., **{'data': ...}}` can also be written as `{..., 'data': ...}`, right?

Reply:
I'm not sure, I'll try that out. Thanks for all of the suggestions.

Reply:
updated